main.go

```diff
@@ -93,8 +104,25 @@ func markPDF(inputPath string, outputPath string, watermark string) error {
 	numPages, err := pdfReader.GetNumPages()
 	fatalIfError(err, fmt.Sprintf("Failed to get PageCount of the source file. [%s]", err))
 
+	// Prepare data to insert into the template.
+	rec := Recipient{
+		Pages:    numPages,
+		Filename: filepath.Base(inputPath[:len(inputPath)-len(filepath.Ext(inputPath))]),
+	}
+	var t *template.Template
+	buf := new(bytes.Buffer)
+	var watermarksIsATemplate bool
+	if !isImageMark{
```

Comment:
- As all these activities are relevant only if `!isImageMark`. So, can we move them inside the condition?
- Also, IMHO, instead of doing it all in `markPDF`, moving the logical portion to a separate method will make it more readable and maintainable.

Reply:
Hey @ajaxray . Thank you so much for taking time to review.

1. May still have to declare them as variables declared & initialized inside a block is scoped ito that block in golang. Example - https://go.dev/play/p/EBhwCgSZSAt
2. I have done some cleanup with this commit and moved applyTemplate to a function https://github.com/AnkurGel/markpdf/commit/b5ad0e90d20cb336c681e5988ca5f815d6332685